crates/core/src/db/datastore/traits.rs

```diff
@@ -297,16 +311,49 @@ pub trait DataRow: Send + Sync {
 pub trait Tx {
     type Tx;
 
+    /// Begins a read-only transaction under the given `workload`.
+    ///
+    /// While this transaction is pending,
+    /// other read-only transactions may be started,
+    /// but new mutable transactions will block until there are no read-only transactions left.
+    ///
+    /// Blocks if a mutable transaction is pending.
```

Comment:
NIT: This doc comment is only applicable to the `LockingTxDatastore`, not to the trait and read transactions in general. I would rather this comment be moved to the appropriate place.